packages/cms/pages/settings/team-members/team-members.tsx

```diff
@@ -68,18 +77,162 @@ const UserRole = styled.span`
+          </EuiFormRow>
+          <EuiFormRow
+            label="Password"
+            fullWidth
+            isInvalid={!!errors}
+            error={errors ? errors[0].message! : ''}
+          >
+            <EuiFieldPassword
+              name="password"
+              fullWidth
+              onChange={e => {
+                setMemberDetails({
+                  ...memberDetails,
+                  password: e.target.value
+                })
+              }}
+            />
+          </EuiFormRow>
+          <EuiFormRow
+            label="Confirm Password"
+            fullWidth
+            isInvalid={!!confirmPasswordError}
+            error={confirmPasswordError}
+          >
+            <EuiFieldPassword
+              name="confirmpassword"
+              fullWidth
+              onChange={e => setConfirmPassword(e.target.value)}
+            />
+          </EuiFormRow>
+          <EuiButton
+            onClick={() => {
+              showChangeMemberRoleModal()
+            }}
+          >
+            Change role
+          </EuiButton>
+        </EuiForm>
+      </EuiFlyoutBody>
+      <EuiFlyoutFooter>
+        <EuiFlexGroup justifyContent="spaceBetween">
+          <EuiFlexItem grow={false}>
+            <EuiButtonEmpty
+              iconType="cross"
+              flush="left"
+              onClick={() => setIsFlyoutVisible(false)}
+            >
+              Cancel
+            </EuiButtonEmpty>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiButton
+              onClick={async () => {
+                if (memberDetails.password != confirmPassword) {
+                  setConfirmPasswordError('passwords do not match')
+                  return
+                }
+                const [data, error] = await window.Tensei.api.patch(
+                  `admin-users/${selectedMember.id}`,
+                  memberDetails
+                )
+                if (!error) {
+                  toast('Edited.', <p>Member data edited successfully.</p>)
+                  setIsFlyoutVisible(false)
+                  getTeamMembers()
+                } else {
+                  setErrors(error.response?.data.errors)
+                }
+              }}
+              fill
+            >
+              Submit
+            </EuiButton>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      </EuiFlyoutFooter>
+    </EuiFlyout>
+  )
+}
 
 export const TeamMembers: FunctionComponent<ProfileProps> = () => {
   const { toast } = useToastStore()
-  const {
-    getAdminUsers,
-    removeUser,
-    getAdminRoles,
-    updateUserRoles
-  } = useAdminUsersStore()
+  const { getAdminUsers, removeUser, getAdminRoles, updateUserRoles } =
+    useAdminUsersStore()
   const [teamMembers, setTeamMembers] = useState<TeamMemberProps[]>([])
   const [roles, setRoles] = useState<any[]>([])
   const [loading, setLoading] = useState(true)
+  const [isFlyoutVisible, setIsFlyoutVisible] = useState(false)
```

Comment:
We should give this a better name so at first glance we know what flyout this state controls. An example is `isEditTeamMemberFlyoutVisible`